tcclient/creds.go

```diff
@@ -7,16 +7,75 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"net/http"
 	"strconv"
 	"strings"
 	"time"
 
 	"github.com/taskcluster/jsonschema2go/text"
 	"github.com/taskcluster/slugid-go/slugid"
+	"github.com/tent/hawk-go"
 )
 
-type Credentials struct {
-	ClientId    string `json:"clientId"`
+type Credentials interface {
+	fmt.Stringer
+	ConfigureAuth(httpRequest *http.Request) error
+	getExtField() (string, error)
+	HawkCredentials() *hawk.Credentials
```

Comment:
Note, I'm not reviewing further to reduce noise... I think this is where we have opportunities for big wins.
Reducing the need for error checking would certainly be a neat bonus to along with this...

Even if we don't use these `SignRequest` and `SignUrl` methods in our code, it means we get errors as soon as possible. That's going to be preferred, always... Rather than getting errors about something bad in configuration, when we start processing tasks for example.